test/resolve-ref.ts

```diff
@@ -1,64 +1,280 @@
+                'should-be-tag',
+                'test-annotated-tag',
+                'test-annotated-tag/packed',
+                'test-tag',
+                'with-slash/should-be-tag'
+            ];
 
             assert.deepStrictEqual(actual, expected);
         });
 
-        it('list tags with oids', async () => {
+        it('tags with oids', async () => {
             const actual = await repo.listTags(true);
             const expected = [
                 { name: 'onmain-tag', oid: '7c2a62cdbc2ef28afaaed3b6f3aef9b581e5aa8e' },
+                { name: 'should-be-tag', oid: '7b84f676f2fbea2a3c6d83924fa63059c7bdfbe2' },
                 { name: 'test-annotated-tag', oid: '56ea7a808e35df13e76fee92725a65a373a9835c' },
-                { name: 'test-tag', oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6' }
+                {
+                    name: 'test-annotated-tag/packed',
+                    oid: '56ea7a808e35df13e76fee92725a65a373a9835c'
+                },
+                { name: 'test-tag', oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6' },
+                {
+                    name: 'with-slash/should-be-tag',
+                    oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6'
+                }
             ];
 
             assert.deepStrictEqual(actual, expected);
         });
     });
+
+    describe('listRemotes()', () => {
+        it('remotes', async () => {
+            const actual = await repo.listRemotes();
+            const expected = ['origin', 'with-slash'];
+
+            assert.deepStrictEqual(actual, expected);
+        });
+    });
+
+    describe('resolveRef()', () => {
+        it('oid into oid', async () => {
+            const oid = '7b84f676f2fbea2a3c6d83924fa63059c7bdfbe2';
+            const actual = await repo.resolveRef(oid);
+
+            assert.strictEqual(actual, oid);
+        });
+
+        describe('ref into oid', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef(testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('refs starting with "ref: " ', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef('ref: ' + testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('throws not found error with correct, but broken refs', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, () => {
+                    assert.rejects(
+                        () => repo.resolveRef(testCase.ref + 'n'),
+                        /ref not found refs\/heads\/mainn/
```

Comment:
Looks like it doesn't work (always positive), since you're passing different refs to `resolveRef()`. Also the message is outdated. Find and fix a mistake in the test source.